src/applications/vaos/package.json

```diff
@@ -0,0 +1,42 @@
+{
+    "name": "@department-of-veterans-affairs/applications-vaos",
+    "version": "1.0.0",
+    "licence": "MIT",
+    "scripts": {
+      "build": "echo Your build script goes here",
+      "test": "echo Your test script goes here",
+      "demo":"echo DEMOING $npm_package_name v $npm_package_version with WORKSPACES"
+    },
+    "dependencies": {
+      "classnames": "^2.3.1",
+      "history-v4": "npm:history@4.9.0",
+      "moment": "~2.29.4",
+      "moment-timezone": "^0.5.34",
+      "prop-types": "^15.6.1",
+      "reselect": "^2.5.4"
+    },
+    "devDependencies": {
+      "@babel/parser": "^7.17.7",
+      "@babel/traverse": "^7.17.3",
+      "@sentry/browser": "^6.13.2",
+      "@testing-library/dom": "^7.26.6",
+      "@testing-library/react": "^11.1.2",
+      "@testing-library/user-event": "^12.2.2",
+      "append-query": "^2.1.1",
+      "chai": "^4.3.4",
+      "glob": "^7.2.0",
+      "just-diff": "^3.1.1",
+      "mockdate": "^3.0.5",
+      "sinon": "^3.2.1"
+    },
+    "peerDependecies":{
+        "react": "*",
+        "react-redux": "*",
+        "react-router": "*" ,
+        "react-router-dom": "*",
+        "redux": "*",
+        "redux-thunk": "*"       
+    },
```

Comment:
What's the policy on deciding what is declared as a dependency, a peer dependency, or not declared at all? For example, [VAOS imports from the `component-library`](https://github.com/department-of-veterans-affairs/vets-website/blob/639f21d9731c24a61f3c0eaa371c529288ec0e63/src/applications/vaos/appointment-list/components/cancel/CancelAppointmentConfirmationModal.jsx#L2) but that isn't represented here in the package manifest.

Reply:
I will add a highlight about this in the docs  
In yarn workspaces,   `"*"`  as a peer dependency actually means use the version installed in the root `package.json` of the repository. if vets-website bumps react, for example, you are bumped right along.  Think of this implementation as the opposite of `nohoist`.

There are some  imports that  were not listed in the  root `package.json`;  these were probably hard coded into the app rather installed with a CLI command (and thus have no current presence  in `node_modules`) At which point the codeowner has to make  decision to not declare it, or declare it as a `nohoist`.

Reply:
I'm confused. `react` is listed as a dependency in the root `package.json` file:

https://github.com/department-of-veterans-affairs/vets-website/blob/8705802bf9d778119bbeb65a0cbd11caeab42d73/package.json#L290

And it is imported in various vaos files:

https://github.com/department-of-veterans-affairs/vets-website/blob/8705802bf9d778119bbeb65a0cbd11caeab42d73/src/applications/vaos/components/AddToCalendar.jsx#L5

And it is also listed as a peer dependency in this PR.

<br/>
<br/>

`@department-of-veterans-affairs/component-library` is also listed as a dependency in the root `package.json` file:

https://github.com/department-of-veterans-affairs/vets-website/blob/8705802bf9d778119bbeb65a0cbd11caeab42d73/package.json#L250

And it is also imported in various files:

https://github.com/department-of-veterans-affairs/vets-website/blob/639f21d9731c24a61f3c0eaa371c529288ec0e63/src/applications/vaos/appointment-list/components/cancel/CancelAppointmentConfirmationModal.jsx#L2

But it _isn't_ listed as a peer dependency in the application. So I'm wondering what the criteria is that determined that react, redux, etc. were to be listed as peer dependencies and other dependencies were not.

Reply:
I see other workspace manifest files list peer dependencies other than react and redux, so it was unclear why an app would be different.

https://github.com/department-of-veterans-affairs/vets-website/blob/8705802bf9d778119bbeb65a0cbd11caeab42d73/src/platform/forms-system/package.json#L62-L68

https://github.com/department-of-veterans-affairs/vets-website/blob/8705802bf9d778119bbeb65a0cbd11caeab42d73/src/platform/site-wide/package.json#L44-L46

Hopefully that helps make the earlier question clearer.

Reply:
Great questions! Just did a quick tutorial to get familiar with yarn workspaces. I also, loaded the PR into a clean repository to verify everything is working. It's working but I am confused about how it's working when none of the component libraries are listed as 'peerDependencies' in vaos/package.json. I expected it to break but it seems to look in the root for the package for missing dependencies regardless of it being listed as a peer dependency. I'm still getting up to speed on this but I'm looking for documentation on the rules. 

I also have the following questions:

- Is VAOS the 1st project to use workspaces? I don't see any other projects listed in root package.json
- How much time do we have to verify this, get team members up to speed, and determine how to manage releases going forward?
- Is the expectation that the version number is bumped each time VAOS is released or is it ignored?

Reply:
@vbahinwillit --
* Nope, check-in had their PR merged in on Tue. They were the first app.
* Our policy is that you have 30 days to approve ( or make changes, the purpose of a workspace is to provide autonomy , after all)  after that the PR is considered stale.
* At the moment, the version number  has no effect as as VSF apps are NOT  being published to NPM, but perhaps it's a good habit to get into.

@bkjohnson 
The dependencies listed are our best assessment of the vaos app imports, part of the function of this PR is to fine tune that, so feel free to list, remove or edit.

A TL;DR on dependencies:
1) During our original set up, we run a [script](https://depo-platform-documentation.scrollhelp.site/developer-docs/yarn-workspace-utility-scripts) that lists ALL imports , and checks against the root node_modules directory  within your application, classifying them as deps, devDeps, or uninstalled . We then sift (eliminating non-module imports)



The docs:

Reply:
I see `@department-of-veterans-affairs/component-library` in the "uninstalled" dictionary of the script output:

![Output of `yarn list-imports:apps --app-folder vaos` with the component-library references highlighted in the "uninstalled" section](https://user-images.githubusercontent.com/2008881/193143984-2e6ff180-3778-4f61-8ad1-63777c07724e.png)

Were some of these uninstalled dependencies overlooked or intentionally excluded from being listed as a peer dependency?

Reply:
@bkjohnson,
Yes, many of the "uninstalled" dependencies may have been overlooked due to ambiguity. 
The script returns  a categorized list of import calls to be used as a general guideline, not a rule.
It functions by scraping all import calls in the directory and comparing them to the list of packages in the root node_modules, this has 2 limitations:
 1) It sometimes confuses regular file imports for modules ( which is what I meant earlier that some sifting is needed)
 2) More to the point of your question,  it can't compare uninstalled modules to the root node_modules so all we know of these were that they are being called somewhere in your directory, but not  if they are active, or needed a specific version, etc. that the code-owner intended.